app/novelty/compute_novelty.py

```diff
@@ -132,175 +286,78 @@ def recency_function_exp(number_of_publ, age_of_oldest_publ, max_number, max_age
     elif np.isnan(age_of_oldest_publ) and not np.isnan(number_of_publ):
-        recency = alter_number_of_publ
+        oldest = alter_number_of_publ
     else:
-        recency = alter_number_of_publ * alter_age_of_oldest_publ
+        oldest = alter_number_of_publ * alter_age_of_oldest_publ
 
-    return recency
+    recency = 1 - oldest
 
+    return recency
 
-async def extracting_drug_fda_publ_date(message, unknown):
+def extracting_publications(message, result):
     """
-    Upon querying, the response is returned as a list containing 10 dictionaries,
-    with each dictionary representing the response from an ARA. The function 'extracting_drug_fda_publ_date'
-    is designed to extract the drug entity name of each edge. It then checks the EPC attributes of each edge
-    to determine the FDA approval status of the drug and the accociated pulications PMID/PMC ID ... .
-    And finally extract the publishing date of the publications to get the oldest (Year) date among them.
-
-    Args:
-        Dictionary: response for a ARA to a query
-
-    Returns:
-        "An DataFrame constructed where each row represents an edge and contains information such as the drug entity
-        name, FDA status of the drug, a list of associated publications, the number of associated publications,
-        and the oldest publication date (year) linked to each drug."
-
+    Function to extract publications information for every result.
+    Four ARAs: ARAX, ARAGORN, BTE and UNSECRET produce results with publications informations
     """
-    attribute_type_id_list_fda = [
-        "biolink:FDA_approval_status",
-        "biolink:FDA_APPROVAL_STATUS",
-    ]
-    attribute_type_id_list_pub = [
-        "biolink:publications",
-        "biolink:Publication",
-        "biolink:publication",
-    ]
-    drug_idx_fda_status = []
-    today = date.today()
-
-    res_chk = 1
-    query_known, query_unknown, query_chk = query_id(message)
-    idi = -1
-    for tmp in unknown:
-        tmp_res = message["results"][tmp]["analyses"][0]["edge_bindings"]
-        for tmp_1 in tmp_res:
-            idi += 1
-            edge_id = message["results"][tmp]["analyses"][0]["edge_bindings"][tmp_1][0][
-                "id"
-            ]
-            edge = message["knowledge_graph"]["edges"][edge_id]
-            if query_chk == 1:
-                if (
-                    "PUBCHEM" in edge["subject"]
-                    or "CHEMBL" in edge["subject"]
-                    or "UNII" in edge["subject"]
-                    or "RXNORM" in edge["subject"]
-                    or "UMLS" in edge["subject"]
-                    or not "MONDO" in edge["subject"]
-                ):
-                    drug_idx = edge["subject"]
+    publications = []
+    for idi, i in enumerate(result['analyses']):
+        result_resource = i['resource_id']
+        edge_keys = list(i['edge_bindings'].keys())
+        if result_resource=="infores:unsecret-agent":
```

Comment:
Can we find a better way to do this? What about the CQS? What about any other ARAs that get added, or ARAs that get updated and publication id locations move slightly? To me, this seems very brittle and it should be a function that looks in all places for any publication info regardless of which ARA it's from.

Reply:
It was my understanding that the code to be pushed is limited to only have the 5 ARAs and not CQS. I specifically targeted the way the publication IDs are stored within each ARA to reduce computation time. Unfortunately, each ARA stores this information differently. The alternative would be to check for all locations across all ARAs which could increase computation time.

Reply:
I think the dynamic function for all ARAs far outweighs the computational overhead of checking all locations. I also think the increase in computation time will be negligible.

Reply:
Got it. I can updated the code to have this added in the next commit.